message.go

```diff
@@ -1,20 +1,54 @@
 package datatransfer
 
 import (
+	"errors"
+	"fmt"
 	"io"
+	"strconv"
+	"strings"
 
 	"github.com/ipfs/go-cid"
 	"github.com/ipld/go-ipld-prime"
 	"github.com/ipld/go-ipld-prime/datamodel"
-	"github.com/libp2p/go-libp2p-core/protocol"
 
 	"github.com/filecoin-project/go-data-transfer/v2/encoding"
 )
 
+type MessageVersion struct {
+	Major uint64
+	Minor uint64
+	Patch uint64
+}
+
+func (mv MessageVersion) String() string {
+	return fmt.Sprintf("%d.%d.%d", mv.Major, mv.Minor, mv.Patch)
+}
+
+// MessageVersionFromString parses a string into a message version
+func MessageVersionFromString(versionString string) (MessageVersion, error) {
+	versions := strings.Split(versionString, ".")
```

Comment:
looks like we have a `stringjoin` representation here, maybe something we can add in later
could also use the new CustomType stuff in bindnode to put `MessageVersion` in the struct itself